include/yaml-cpp/node/detail/node.h

```diff
@@ -160,8 +166,10 @@ class node {
 
  private:
   shared_node_ref m_pRef;
-  typedef std::set<node*> nodes;
+  typedef std::set<node*, less> nodes;
   nodes m_dependencies;
+  size_t m_index;
+  static size_t m_amount;
```

Comment:
This isn't threadsafe.

Reply:
How about this?

Reply:
Almost an anniversary.

Reply:
To be fair, `std::set` also isn't thread-safe.

Reply:
de-vri-es, that's true. Any ideas?

Reply:
I don't think it makes sense to try to make `Node` thread safe. I think it is better to just document the fact that it isn't and require the user to do synchronization. Doing it inside `Node` would add a lot of complexity and hurt performance, also in cases where it isn't needed.